plugins/tc/tbf/main.go

```diff
@@ -0,0 +1,208 @@
+	Burst int `json:"burst"` //Bandwidth burst in Kb for traffic through container. 0 for no limit. If burst is set, rate must also be set
+}
+
+// parseConfig parses the supplied configuration (and prevResult) from stdin.
+func parseConfig(stdin []byte) (*PluginConf, error) {
+	conf := PluginConf{}
+
+	if err := json.Unmarshal(stdin, &conf); err != nil {
+		return nil, fmt.Errorf("failed to parse network configuration: %v", err)
+	}
+
+	// Parse previous result. Remove this if your plugin is not chained.
+	if conf.RawPrevResult != nil {
+		resultBytes, err := json.Marshal(conf.RawPrevResult)
+		if err != nil {
+			return nil, fmt.Errorf("could not serialize prevResult: %v", err)
+		}
+		res, err := version.NewResult(conf.CNIVersion, resultBytes)
+		if err != nil {
+			return nil, fmt.Errorf("could not parse prevResult: %v", err)
+		}
+		conf.RawPrevResult = nil
+		conf.PrevResult, err = current.NewResultFromResult(res)
+		if err != nil {
+			return nil, fmt.Errorf("could not convert result to current version: %v", err)
+		}
+	}
+	// End previous result parsing
+
+	switch {
+	case conf.Burst < 0 || conf.Rate < 0:
+		return nil, fmt.Errorf("rate and burst must be a positive integer")
+	case conf.Burst == 0 && conf.Rate != 0:
+		return nil, fmt.Errorf("if rate is set, burst must also be set")
+	case conf.Rate == 0 && conf.Burst != 0:
+		return nil, fmt.Errorf("if burst is set, rate must also be set")
+	}
+
+	return &conf, nil
+
+}
+
+func getIfbDeviceName(networkName string, containerId string) (string, error) {
+	hash := sha1.New()
+	_, err := hash.Write([]byte(networkName + containerId))
+	if err != nil {
+		return "", err
+	}
+
+	return fmt.Sprintf("%x", hash.Sum(nil))[:15], nil
+}
+
+func getMTU(deviceName string) (int, error) {
+	link, err := netlink.LinkByName(deviceName)
+	if err != nil {
+		return -1, err
+	}
+
+	return link.Attrs().MTU, nil
+}
+
+// cmdAdd is called for ADD requests
+func cmdAdd(args *skel.CmdArgs) error {
+	conf, err := parseConfig(args.StdinData)
+	if err != nil {
+		return err
+	}
+
+	//no traffic shaping was requested, so just no-op and quit
+	if conf.Rate == 0 && conf.Burst == 0 {
+		return types.PrintResult(conf.PrevResult, conf.CNIVersion)
+	}
+
+	if conf.PrevResult == nil {
+		return fmt.Errorf("must be called as chained plugin")
+	}
+
+	containerIPConfigs := make([]*current.IPConfig, 0, len(conf.PrevResult.IPs))
+
+	for _, ip := range conf.PrevResult.IPs {
+		if ip.Interface == nil {
+			continue
+		}
+		intIdx := *ip.Interface
+		// Every IP is indexed in to the interfaces array, with "-1" standing
+		// for an unknown interface (which we'll assume to be Container-side
+		// Skip all IPs we know belong to an interface with the wrong name.
+		if intIdx >= 0 && intIdx < len(conf.PrevResult.Interfaces) && conf.PrevResult.Interfaces[intIdx].Name != args.IfName {
+			continue
+		}
+		containerIPConfigs = append(containerIPConfigs, ip)
+	}
+
+	if len(containerIPConfigs) == 0 {
+		return fmt.Errorf("got no container IPs")
```

Comment:
Why bother with this check?  I don't see `containerIPConfigs` used later in this function.